test/calendar_test.exs

```diff
@@ -22,6 +22,30 @@ if Code.ensure_loaded?(Calendar) do
       assert [[~T[01:02:03.123456]]] = query("SELECT time '01:02:03.123456'", [])
     end
 
+    test "decode timetz", context do
+      assert [[~T[00:00:00.000000]]] = query("SELECT time with time zone '00:00:00 UTC'", [])
+      assert [[~T[01:02:03.000000]]] = query("SELECT time with time zone '01:02:03 UTC'", [])
+      assert [[~T[23:00:00.000000]]] = query("SELECT time with time zone '00:00:00+1'", [])
+
+      assert :ok = query("SET SESSION TIME ZONE UTC", [])
+
+      assert [[~T[23:59:59.000000]]] = query("SELECT time with time zone '23:59:59'", [])
+      assert [[~T[12:05:06.000000]]] = query("SELECT time with time zone '04:05:06 PST'", [])
+
+      assert [[~T[00:00:00.123000]]] = query("SELECT time with time zone '00:00:00.123'", [])
+      assert [[~T[00:00:00.123456]]] = query("SELECT time with time zone '00:00:00.123456 UTC'", [])
+      assert [[~T[01:02:03.123456]]] = query("SELECT time with time zone '01:02:03.123456'", [])
+
+      assert [[~T[01:02:03.123456]]] = query("SELECT time with time zone '16:01:03.123456+1459'", [])
+
+      assert [[~T[16:01:03.123456]]] = query("SELECT time with time zone '01:02:03.123456-1459'", [])
+
+      assert :ok = query("SET SESSION TIME ZONE +1", [])
+
+      assert [[~T[12:05:06.000000]]] = query("SELECT time with time zone '04:05:06 PST'", [])
```

Comment:
I like that this example is copied from before the `SET SESSION TIME ZONE +1` line.

Even though PST is one thing in postgres, outside of postgres "PST" is ambiguous, so I would add another non-UTC example that uses a numeric offset instead of a three letter code. For instance a line like this both before and after the `SET SESSION TIME ZONE +1` line:
`assert [[~T[12:05:06.000000]]] = query("SELECT time with time zone '04:05:06-8'", [])`